demos/smart_classroom_demo_gapi/main.cpp

```diff
@@ -0,0 +1,561 @@
+        const auto top_action_id = actions_type == TOP_K
+                                   ? std::distance(actions_map.begin(), find(actions_map.begin(), actions_map.end(), FLAGS_top_id))
+                                   : -1;
+        if (actions_type == TOP_K && (top_action_id < 0 || top_action_id >= static_cast<int>(actions_map.size()))) {
+            slog::err << "Cannot find target action: " << FLAGS_top_id << slog::endl;
+            return 1;
+        }
+
+        slog::info << "Reading video '" << video_path << "'" << slog::endl;
+        slog::info << "Loading Inference Engine" << slog::endl;
+        Core ie;
+        std::vector<std::string> devices = {FLAGS_d_act, FLAGS_d_fd, FLAGS_d_lm,
+                                            FLAGS_d_reid};
+        std::set<std::string> loadedDevices;
+        slog::info << "Device info: " << slog::endl;
+        for (const auto &device : devices) {
+            if (loadedDevices.find(device) != loadedDevices.end())
+                continue;
+            std::cout << printable(ie.GetVersions(device)) << std::endl;
+            loadedDevices.insert(device);
+        }
+
+        cv::Mat frame;
+        cv::VideoCapture cap;
+        video_path != "cam" ? cap.open(video_path) : cap.open(0);
+        if (!cap.read(frame)) {
+            GAPI_Assert(false && "Couldn't grab the very first frame");
+        }
+
+        /** NOTE: Empty strings. Need default constructor for cv::gapi::ie::Params */
+        cv::gapi::ie::Params<nets::FaceDetector> det_net({}, {}, {});
+        cv::gapi::ie::Params<nets::LandmarksDetector> landm_net({}, {}, {});
+        cv::gapi::ie::Params<nets::FaceReidentificator> reident_net({}, {}, {});
+        cv::gapi::ie::Params<nets::PersonDetActionRec> action_net({}, {}, {});
+
+        ActionDetectorConfig action_config;
+        ActionDetectionKernelInput ad_kernel_input;
+        if (!ad_model_path.empty()) {
+            // Load action detector
+            action_config.net_with_six_actions = IsNetForSixActions(ad_model_path);
+            action_config.detection_confidence_threshold = static_cast<float>(FLAGS_t_ad);
+            action_config.action_confidence_threshold = static_cast<float>(FLAGS_t_ar);
+            action_config.num_action_classes = actions_map.size();
+            action_config.input_height = cap.get(cv::CAP_PROP_FRAME_HEIGHT);
+            action_config.input_width = cap.get(cv::CAP_PROP_FRAME_WIDTH);
+            ad_kernel_input.ptr.reset(new ActionDetection(action_config));
+            std::array<std::string, 7> outputBlobList;
+            outputBlobList = action_config.net_with_six_actions
+                ? outputBlobList = action_detector_6
+                : outputBlobList = action_detector_5;
+            action_net = cv::gapi::ie::Params<nets::PersonDetActionRec>{
+                ad_model_path,
+                GetBinPath(ad_model_path),
+                FLAGS_d_act,
+            }.cfgOutputLayers(outputBlobList);
+        }
```

Comment:
I'd move all of this out of `main()`. It makes reading and understanding the `main()` flow really hard when it is that long.

Reply:
Moved to .hpp. Added comments for main steps.

Reply:
Not sure what is moved where but the code producing the network packages based on options is still there.

May not be a problem now, though.